src/stack_limiter/max_height.rs

```diff
@@ -113,44 +118,40 @@ impl Stack {
 				return if top_frame.is_polymorphic {
 					Ok(())
 				} else {
-					return Err("trying to pop more values than pushed")
+					return Err(anyhow!("trying to pop more values than pushed"))
 				}
 			}
 		}
 
-		self.height = self.height.checked_sub(value_count).ok_or("stack underflow")?;
+		self.height =
+			self.height.checked_sub(value_count).ok_or_else(|| anyhow!("stack underflow"))?;
 
 		Ok(())
 	}
 }
 
-/// This function expects the function to be validated.
-pub fn compute(func_idx: u32, module: &elements::Module) -> Result<u32, &'static str> {
-	use parity_wasm::elements::Instruction::*;
-
-	let func_section = module.function_section().ok_or("No function section")?;
-	let code_section = module.code_section().ok_or("No code section")?;
-	let type_section = module.type_section().ok_or("No type section")?;
-
+/// This function expects the function to be validated. func_idx have sub import funcs num
```

Comment:
Ideally, this can be done in an early pass right ? (Once this is implemented https://github.com/filecoin-project/ref-fvm/issues/1038#issuecomment-1295119234 )